### ISSUE TITLE
Expose Availability Zone ID in addition to name

### ISSUE THREAD

**Issue by sergey-kalin** [danfromtitan/envars-from-node-labels#17]
Currently AZ name is returned e.g. `us-east-1c`

`NODE_TOPOLOGY_KUBERNETES_IO_ZONE=us-east-1c`

There is case with Amazon managed Kafka service (MSK) when we can consume data from Kafka broker deployed same AZ with container - [docs](https://aws.amazon.com/blogs/big-data/reduce-network-traffic-costs-of-your-amazon-msk-consumers-with-rack-awareness/)

In order to make it work both MSK and container should know AZ Identifier (not AZ name). As in particular MSK case Kafka brokers can only know AZ ID (provided by AWS).

AZ ID usually look something like `use1-az3`. E.g. in my case AZ with name `us-east-1e` has ID  `use1-az3`.

Sample call to AWS CLI
```
# aws ec2 describe-availability-zones
{
    "AvailabilityZones": [
..
        {
...
            "RegionName": "us-east-1",
            "ZoneName": "us-east-1e",
            "ZoneId": "use1-az3",
...
```

Basically I'd like to have `ZoneId` value too, not only `ZoneName` currently provided.

Without it I have to make call to AWS API from container to resolve name to id, that kills all the elegance of admission controller usage :)

**Comment by danfromtitan** [comment]
This is a use case on the very edge of the implementation design, mainly because zone IDs are specific to a subscription account, not so much to an EKS node. I understand the usefulness of it though, so I'll add a zone ID label next to the zone name.

What is the API call you're making now to determine the zone ID @sergey-kalin ?

**Comment by sergey-kalin** [comment]
@danfromtitan I'm using Java version of `aws ec2 describe-availability-zones` actually, just getting all zones visible to container and filtering one needed. 

There is same method for Go SDK - https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#EC2.DescribeAvailabilityZones

Please mind it requires `ec2:DescribeAvailabilityZones` permission granted through e.g. service account

**Comment by danfromtitan** [comment]
@sergey-kalin, I was looking at a proper way to solve this, because like I said earlier it doesn't feel the right solution making calls into AWS control plane from a webhook.

There is a way to add additional node labels from the worker bootstrap script (i.e. /etc/eks/bootstrap.sh). The way you go about this is to [use kubelet-extra-args ](https://github.com/awslabs/amazon-eks-ami/issues/67) with the bootstrap script.

So you're going to have a little script that would make a [curl request into AWS metadata plane](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html) to find the zone ID mapping, then create new labels with kubelet-extra-args in the same script. You wrap it all together by running the script from a terraform launch template at EKS cluster creation time:

```
{
  filename     = "custom-labels.sh"
  content_type = "text/cloud-boothook"
  vars         =  {}
}
```
